src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -516,6 +520,77 @@ Namespace Microsoft.CodeAnalysis.Operations
             End If
         End Function
 
+        Private Enum NestedType
+            None
+            DelegateCreation
+            ExpressionTree
+            Conversion
+        End Enum
+
+        Private Shared Function GetNestedInfo(operand As BoundNode, conversionKind As ConversionKind, conversionSyntax As SyntaxNode, targetType As TypeSymbol
+                                                                         ) As (NestedType As NestedType, NestedConversionStruct As Conversion)
+            If conversionKind = ConversionKind.Identity AndAlso operand.Kind = BoundKind.Parenthesized Then
+                Dim nestedOperand = operand
+                Do
+                    nestedOperand = DirectCast(nestedOperand, BoundParenthesized).Expression
+                Loop While nestedOperand.Kind = BoundKind.Parenthesized
+
+                Select Case nestedOperand.Kind
+                    Case BoundKind.Conversion, BoundKind.DirectCast, BoundKind.TryCast
+                        Dim nestedConversion = DirectCast(nestedOperand, BoundConversionOrCast)
+                        If nestedConversion.Type <> targetType Then
+                            Return (NestedType.None, NestedConversionStruct:=Nothing)
+                        End If
+
+                        If IsDelegateCreation(nestedConversion.ConversionKind, conversionSyntax, nestedConversion.Operand, targetType) Then
+                            Return (NestedType.DelegateCreation, CreateConversion(nestedConversion))
+                        ElseIf (nestedConversion.ConversionKind And ConversionKind.ConvertedToExpressionTree) = ConversionKind.ConvertedToExpressionTree Then
+                            Return (NestedType.ExpressionTree, CreateConversion(nestedConversion))
+                        ElseIf nestedConversion.Operand.Kind = BoundKind.Conversion OrElse
+                               nestedConversion.Operand.Kind = BoundKind.TryCast OrElse
+                               nestedConversion.Operand.Kind = BoundKind.DirectCast OrElse
+                               nestedConversion.Operand.Kind = BoundKind.AddressOfOperator Then
+                            ' If there is another conversion under this nested conversion, when we assume that this conversion should be pruned like in other cases
```

Comment:
>we assume that this conversion should be pruned like in other cases [](start = 96, length = 67)

Why is it safe to make this assumption? What if the conversion was explicitly present in code?